gaming/src/golang/item-service/main_test.go

```diff
@@ -0,0 +1,450 @@
+//go:build integration
+
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+
+	"cloud.google.com/go/spanner"
+	database "cloud.google.com/go/spanner/admin/database/apiv1"
+	instance "cloud.google.com/go/spanner/admin/instance/apiv1"
+	"embed"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/googlecloudplatform/cloud-spanner-samples/gaming-item-service/models"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	databasepb "google.golang.org/genproto/googleapis/spanner/admin/database/v1"
+	instancepb "google.golang.org/genproto/googleapis/spanner/admin/instance/v1"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+//go:embed test_data/schema.sql
+var SCHEMAFILE embed.FS
+
+var TESTNETWORK = "game-sample-test"
+
+// These integration tests run against the Spanner emulator. The emulator
+// must be running and accessible prior to integration tests running.
+
+type Emulator struct {
+	testcontainers.Container
+	Endpoint string
+	Project  string
+	Instance string
+	Database string
+}
+
+type Service struct {
+	testcontainers.Container
+	Endpoint string
+}
+
+func teardown(ctx context.Context, emulator *Emulator, service *Service) {
+	emulator.Terminate(ctx)
+	service.Terminate(ctx)
+}
+
+func setupSpannerEmulator(ctx context.Context) (*Emulator, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        "gcr.io/cloud-spanner-emulator/emulator:1.4.2",
```

Comment:
Maybe use tag `latest` instead of a fixed version? That would catch any regressions caused by the emulator, which users are likely to run into, as they will most probably use the latest version.

Reply:
:+1: I like that better. Thanks!